src/utils/with-react.js

```diff
@@ -132,11 +132,8 @@ const distributeProperties = (properties, componentClass) => {
       case Array:
       case Object:
       case Function:
-        map = props;
-        break;
       case Boolean:
-        map = attrs;
-        value = value ? '' : null; // ''/null: canonicalize Boolean values s.t. val(...) sets or removes the attribute
+        map = props;
```

Comment:
No need for a switch-case anymore. A simple if should be enough?!

Reply:
As discussed bilaterally, this has its own downsides (repeated type === ...), so I prefer to keep it as-is.